log/log.go

```diff
@@ -130,58 +165,75 @@ func LogLineChecksum(line string) string {
 	return base64.RawURLEncoding.EncodeToString(buf)
 }
 
-// Log the provided message at the appropriate level, writing to
+// logAtLevel logs the provided message at the appropriate level, writing to
 // both stdout and the Logger
 func (w *bothWriter) logAtLevel(level syslog.Priority, msg string) {
-	var prefix string
 	var err error
 
-	const red = "\033[31m\033[1m"
-	const yellow = "\033[33m"
-
 	// Since messages are delimited by newlines, we have to escape any internal or
 	// trailing newlines before generating the checksum or outputting the message.
 	msg = strings.Replace(msg, "\n", "\\n", -1)
 	msg = fmt.Sprintf("%s %s", LogLineChecksum(msg), msg)
 
+	w.Lock()
+	defer w.Unlock()
+
 	switch syslogAllowed := int(level) <= w.syslogLevel; level {
 	case syslog.LOG_ERR:
 		if syslogAllowed {
 			err = w.Err(msg)
 		}
-		prefix = red + "E"
 	case syslog.LOG_WARNING:
 		if syslogAllowed {
 			err = w.Warning(msg)
 		}
-		prefix = yellow + "W"
 	case syslog.LOG_INFO:
 		if syslogAllowed {
 			err = w.Info(msg)
 		}
-		prefix = "I"
 	case syslog.LOG_DEBUG:
 		if syslogAllowed {
 			err = w.Debug(msg)
 		}
-		prefix = "D"
 	default:
 		err = w.Err(fmt.Sprintf("%s (unknown logging level: %d)", msg, int(level)))
 	}
 
 	if err != nil {
-		fmt.Fprintf(os.Stderr, "Failed to write to syslog: %s (%s)\n", msg, err)
+		fmt.Fprintf(os.Stderr, "Failed to write to syslog: %d %s (%s)\n", int(level), msg, err)
 	}
 
-	var reset string
-	if strings.HasPrefix(prefix, "\033") {
-		reset = "\033[0m"
-	}
+	w.stdoutWriter.logAtLevel(level, msg)
+}
+
+// logAtLevel logs the provided message to stdout, or stderr if it is at Warning or Error level.
+func (w *stdoutWriter) logAtLevel(level syslog.Priority, msg string) {
+	if int(level) <= w.level {
+		output := w.stdout
+		if int(level) <= int(syslog.LOG_WARNING) {
+			output = w.stderr
+		}
+
+		var color string
+		var reset string
+
+		const red = "\033[31m\033[1m"
+		const yellow = "\033[33m"
+
+		if w.isatty {
+			if int(level) == int(syslog.LOG_WARNING) {
+				color = yellow
+				reset = "\033[0m"
+			} else if int(level) <= int(syslog.LOG_ERR) {
+				color = red
+				reset = "\033[0m"
+			}
+		}
 
-	if int(level) <= w.stdoutLevel {
-		if _, err := fmt.Fprintf(w.stdout, "%s%s %s %s%s\n",
-			prefix,
-			w.clk.Now().Format("150405"),
+		if _, err := fmt.Fprintf(output, "%s%s %d %s %s%s\n",
+			color,
```

Comment:
I think the tests are complaining about a syntax error here where the `if int(level) <= w.stdoutLevel` conditional got dropped.